src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/XamlIlBindingPathHelper.cs

```diff
@@ -240,6 +244,7 @@ private static XamlIlBindingPathNode TransformBindingPath(AstTransformationConte
                     case TemplatedParentBindingExpressionNode templatedParent:
                         var templatedParentField = context.GetAvaloniaTypes().StyledElement.GetAllFields()
                             .FirstOrDefault(f => f.IsStatic && f.IsPublic && f.Name == "TemplatedParentProperty");
+                        nodes.Add(new SelfPathElementNode(selfType));
                         nodes.Add(new XamlIlAvaloniaPropertyPropertyPathElementNode(
                             templatedParentField,
                             templatedParent.Type));
```

Comment:
Previously compiled generated `{Binding TemplatedParent.Property}` code, when it should `{Binding $self.TemplatedParent.Property}`